File: Introibo/Data/MissalSection.swift
import Foundation

// Matches Introibo/Resources/missal.json.
// 13 sections of the Ordinary of the Mass, in order.

struct MissalSection: Identifiable, Decodable, Hashable {
    let slug: String
    let label: String?
    let title: String
    // Vernacular fields are `var` so the Spanish overlay can rewrite them
    // in place (ContentStore.applyVernacularOverlay).
    var english: String?
    var body: [Line]

    var id: String { slug }

    struct Line: Decodable, Hashable {
        var lat: String
        var eng: String
        var rubric: String? = nil
    }
}

File: android/app/src/main/java/com/lampstandhq/introibo/data/model/MissalSection.kt
package com.lampstandhq.introibo.data.model

import kotlinx.serialization.Serializable

/**
 * Matches missal.json.
 * 13 sections of the Ordinary of the Mass, in order.
 */
@Serializable
data class MissalSection(
    val slug: String,
    val label: String? = null,
    val title: String,
    val english: String? = null,
    val body: List<Line>,
) {
    @Serializable
    data class Line(
        val lat: String,
        val eng: String,
        val rubric: String? = null,
    )
}
